AndrewTehan/hometask-3/rainfall_statistics.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# this class is used for calculate mean and variance values for town rainfall
+class RainfallStatistics
+  attr_reader :town, :data
+
+  def initialize(town, data)
+    @town = town
+    @data = data
+  end
+
+  def mean
+    return -1 if missing_data?
+
+    average(extract_statistic_values)
+  end
+
+  def variance
+    return -1 if missing_data?
+
+    calculate_variance(calculate_mean_differences)
+  end
+
+  private
+
+  def extract_statistic_values
+    @extract_statistic_values ||= find_town_statistics.scan(/[0-9]*\.[0-9]*/).map(&:to_f)
```

Comment:
Please have a look at https://rubular.com/ It will help you use simple and elegant concepts of writing regular expressions. Instead of writing `[0-9]`, you can replace it with `\d`. What do you think about it? is it nicer a bit, nuh? 😎😉✌🏻

![Screen Shot 2021-11-15 at 1 00 02 PM](https://user-images.githubusercontent.com/31599278/141764713-bb1f7104-e4e1-4ecb-a102-c3f024626800.png)
![Screen Shot 2021-11-15 at 12 59 41 PM](https://user-images.githubusercontent.com/31599278/141764719-158119d0-c6e1-4c9e-a556-c565844f570c.png)